admin/class-boldgrid-backup-admin-db-dump.php

```diff
@@ -154,47 +154,49 @@ public function dump( $file ) {
 	public function get_connection_string( $db_host = null, $db_name = null ) {
 		$params = array();
 
+        $has_socket = false;
+
 		// Configure parameters passed in.
 		$db_name = empty( $db_name ) ? DB_NAME : $db_name;
 		$db_host = empty( $db_host ) ? DB_HOST : $db_host;
 		$db_host = explode( ':', $db_host );
 
+        if ( 'sock' === pathinfo( $db_host[0], PATHINFO_EXTENSION ) || 'sock' === pathinfo( $db_host[1], PATHINFO_EXTENSION ) ) {
```

Comment:
Instead of custom logic, could we use https://developer.wordpress.org/reference/classes/wpdb/parse_db_host/ ?